### PR TITLE
Add storage backends

### PR DESCRIPTION
**Pull request by chivay**
Extract logic used for handling ZFS/QCOW2 setups into uniform interface

### CODE REVIEW

**Comment by chivay** [comment]
I don't consider it fully finished, but I'd like a round of review to see what you think.

**Comment by BonusPlay** [comment]
I like it. Maybe we could change `install_info.storage_backend` to `Literal["zfs", "qcow2"]` or `class BackendTypes(str, Enum)` :thinking:

File: drakrun/drakrun/storage.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                        shlex.quote(disk_size),
+                    ]
+                ),
+                shell=True,
+            )
+        except subprocess.CalledProcessError:
+            raise RuntimeError("Failed to create a new volume using qemu-img.")
+
+    def snapshot_vm0_volume(self):
+        # We'll be using vm-0.img as backing storage
```
Comment (BonusPlay) [inline drakrun/drakrun/storage.py:195]: I'd add a log here (might come in handy in the future while debugging).

Comment (chivay) [reply drakrun/drakrun/storage.py:195]: +1 to that, however how do we do it cleanly with karton? AFAIR `import logging` won't work. Or will it?

Comment (BonusPlay) [reply drakrun/drakrun/storage.py:195]: I think you can do `logging.getLogger("karton")` (or w/e kartonik you want).

Comment (chivay) [reply drakrun/drakrun/storage.py:198]: OK, leaving that for now. Added a TODO.

**Comment by chivay** [comment]
Tested and confirmed working with QCOW2

File: drakrun/drakrun/storage.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            # TODO: this assumes /dev/nbd0 is free
+            subprocess.check_output(
+                f"qemu-nbd -c /dev/nbd0 --read-only {os.path.join(LIB_DIR, 'volumes', 'vm-0.img')}",
+                shell=True,
+            )
+        except subprocess.CalledProcessError:
+            raise RuntimeError("Failed to connect QCOW2 file to /dev/nbd0")
+
+        # we mount 2nd partition, as 1st partition is windows boot related and 2nd partition is C:\\
+        yield "/dev/nbd0p2"
```
Comment (icedevml) [inline drakrun/drakrun/storage.py:242]: There is no guarantee at all that this will be 2nd partition, but that's just popular.

Comment (BonusPlay) [reply drakrun/drakrun/storage.py:242]: Yes, there's also no guarantee that `ndb0` will be free to use, both of those are just assumptions.

**Pull request merged**